pinot-core/src/main/java/com/linkedin/pinot/core/segment/index/ColumnMetadata.java

```diff
@@ -157,9 +164,27 @@ public static ColumnMetadata fromPropertiesConfiguration(String column, Properti
       }
     }
 
+
+    String partitionFunctionString = config.getString(getKeyFor(column, V1Constants.MetadataKeys.Column.PARTITION_FUNCTION));
```

Comment:
I agree that it is more likely that partitioning is performed on a composite key, but supporting that would require a lot of changes to existing code. Until then I'd lean towards, allowing partitioning on multiple columns, as it has added benefit of pruning when any of the partitioned columns has predicates (as opposed to all of the columns in composite partitioning).